data/dataset.py

```diff
@@ -0,0 +1,173 @@
+import os.path
+import itertools
+
+import torch
+import pandas as pd
+import fortepyan as ff
+from tqdm import tqdm
+from datasets import load_dataset
+
+from data.quantizer import MidiQuantizer
+from data.tokenizer import Tokenizer, VelocityTokenizer
+from data.prepare_dataset import process_record, unprocessed_samples
+
+
+class TokenizedMidiDataset:
+    def __init__(
+        self,
+        split: str = "train",
+        n_dstart_bins: int = 3,
+        n_duration_bins: int = 3,
+        n_velocity_bins: int = 3,
+        sequence_len: int = 128,
+        device: str = "cpu",
+    ):
+        self.split = split
+        self.sequence_len = sequence_len
+        self.device = device
+
+        self.quantizer = MidiQuantizer(
+            n_dstart_bins=n_dstart_bins,
+            n_duration_bins=n_duration_bins,
+            n_velocity_bins=n_velocity_bins,
+        )
+
+        self.tokenizer_src = Tokenizer(keys=["pitch", "dstart_bin", "duration_bin", "velocity_bin"])
+        self.tokenizer_tgt = VelocityTokenizer()
+
+        self.dataset = load_dataset(path="roszcz/maestro-v1", split=split)
+
+        self.src_vocab, self.tgt_vocab = self._build_vocab()
+
+        self.processed_records, self.unprocessed_records = self.load_dataset()
+        self.samples = self.load_samples()
+
+    def load_samples(self) -> list[tuple[list[int], list[int]]]:
+        path = (
+            f"data/datasets/samples-"
```

Comment:
```suggestion
            f"tmp/datasets/samples-"
```

There's code in the `datasets` folder, and we shouldn't mix storage space with code space :)